js/dialogcards.js

```diff
@@ -853,6 +1090,134 @@ H5P.Dialogcards = (function ($, Audio, JoubelUI) {
     }
   };
 
+  /**
+   * Task is finished.
+   */
+
+  C.prototype.finishedScreen = function () {
+    var self = this;
+
+    // Display task finished feedback message.
+    $feedback = $('<div class="h5p-dialogcards-feedback">' + self.params.finished + '</div>').appendTo(self.$inner);
+
+    // Display reset button to enable user to do the task again.
+    self.$resetTaskButton = JoubelUI.createButton({
+      'class': 'h5p-dialogcards-reset',
+      'title': self.params.retry,
+      'html': self.params.retry
+    }).click(function () {
+      self.resetTask();
+    }).appendTo(self.$inner);
+  }
+
+
+  /**
+   * Remove card from DOM and from cards stack after user has checked the "gotit" button.
+   */
+
+  C.prototype.gotIt = function ($card) {
+    var self = this;
+    self.triggerXAPI('interacted');
+    var index = $card.index();
+    this.taskFinished = false;
+    // Mark current card with a 'gotitdone' class.
+    self.$current.addClass('h5p-dialogcards-gotitdone');
+
+    // Move to next card if exists.
+    var $nextCard = self.$current.next('.h5p-dialogcards-cardwrap');
+    var $prevCard = self.$current.prev('.h5p-dialogcards-cardwrap');
+
+    if ($nextCard.length) {
+        var i = 1;
+        self.nextCard(i);
+    } else if ($prevCard.length) { // No next card left - go to previous.
+        self.prevCard();
+    } else { // No cards left: task is finished.
+      self.progress = -1;
+
+      // Removes all these elements.
+      $('.h5p-dialogcards-cardwrap-set, .h5p-dialogcards-footer', self.$inner).remove();
+
+      // Display task finished feedback message.
+      self.finishedScreen();
+
+      // Display reset (retry) button.
+      self.$inner.find('.h5p-dialogcards-reset').removeClass('h5p-dialogcards-disabled');
+      self.taskFinished = true;
+      self.triggerXAPIScored(1, 1, 'completed');
+    }
+
+    // Now remove the current 'gotitdone' card from the cards and cardOrder arrays.
+      self.dialogs.splice(index, 1);
+      // TODO if exists !
+      self.cardOrder.splice(index, 1);
+
+    if (!self.taskFinished) {
+      // Remove the 'gotitdone' card from DOM
+      var el = '.h5p-dialogcards-gotitdone';
+      $( el ).remove();
```

Comment:
It is strange that it didn't fail. I'm sure there must be scenarios where this causes a problem.